ptbcontrib/chat_to_link/README.md

```diff
@@ -0,0 +1,22 @@
+# Get Joinable Link from Chat
+
+Provides a method to get a joinable link from a `telegram.Chat` object. To do so, it tries
+to get a link in the following order:
+1. Chat's username (`chat.username`).
+2. Chat's invite link (`chat.invite_link`).
+3. Chat's invite link from bot (`bot.get_chat.invite_link`)
+4. Create invite link if `member_limit` or `expire_date` is passed
+   (`bot.create_chat_invite_link`).
```

Comment:
Then we'd need an extra parameter like `export_primary_link=False` to check what kind of link the user wants (primary or not). What shall we call the parameter?

Reply:
uhm, would this actually make a difference? The rights for both method remain the same, no? If you can call one, you can call the other. So I would just decide to use one and be done with it. No need to overwhelm the user with decisions if you can do them for them.

Reply:
So which one should we use? `bot.export_chat_invite_link` creates a new primary link and revokes the other one, whilst `bot.create_chat_invite_link` only creates a permanent non-primary link.
I also had another question: Is it possible for `bot.get_chat().invite_link` to be None and then `bot.export_chat_invite_link` actually return a link? How can the bot export a new link but not have one in the first place?

Reply:
> How can the bot export a new link but not have one in the first place?

If the bot didn't export a link yet, it will be None. Once it exports, it will be returned in get_chat. Thats why I would vote for using that anyway, it saves one API request if I see the logic + remember telegrams logic correctly

Reply:
I thought of that too, but that way we would replace the primary invite link if there's one. So should I remove `bot.get_chat`?

Reply:
> but that way we would replace the primary invite link if there's one

No, if that happens, `if bot_chat.invite_link:` is True

Reply:
Which method should we use for creating a new link? `bot.create_chat_invite_link` or `bot.export_invite_link`?

Reply:
bot.export_chat_invite_link will save API calls, wont it?

Reply:
I meant after checking `bot.get_chat` which returns None. So this is what the function looks like when we choose one of the methods:
```python
...
bot_chat = bot.get_chat(chat.id)
if bot_chat.invite_link:
    return bot_chat.invite_link

try:
    return chat.export_invite_link()
except BadRequest as exc:
    if exc.message == "Not enough rights to manage chat invite link":
        return ""
    raise exc
```

Reply:
yeah, but if you use `create_invite_link`, it wont show up in `bot_chat`, will it? Because `bot_chat.invite_link` will only be the primary invite link. so if you would use `create_invite_link`, you would need to store the link locally (which is also cool but not your original design), in order to not generate a new one if the same chat gets passed. But if you use `export`, it will show up in `get_chat`

Reply:
Yes, but `export` will revoke the current primary link. Is that okay?

Reply:
It wont revoke it if it already exists since then you return from the function. Thats why you call `get_chat` before :P

Reply:
Oh, you're right. I totally missed that :D